src/BlockTypes/ProductGalleryPager.php

```diff
@@ -0,0 +1,132 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+/**
+ * ProductGalleryPager class.
+ */
+class ProductGalleryPager extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-gallery-pager';
+
+	/**
+	 * It isn't necessary register block assets because it is a server side block.
+	 */
+	protected function register_block_type_assets() {
+		return null;
+	}
+
+	/**
+	 *  Register the context
+	 *
+	 * @return string[]
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'productGalleryClientId', 'pagerDisplayMode' ];
+	}
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$pager_display_mode = isset( $block->context['pagerDisplayMode'] ) ? $block->context['pagerDisplayMode'] : '';
```

Comment:
To keep consistency with the code line below `$classname = $attributes['className'] ?? '';` , I'd use the following syntax:
```suggestion
		$pager_display_mode = $block->context['pagerDisplayMode'] ?? '';

```

Reply:
Done!